Documentation/compatibility/winforms-accessibility-changes-471.md

```diff
@@ -0,0 +1,112 @@
+      </runtime>
+    </configuration>
+
+If an application targets version 4.7.1 or above and has preserve the legacy accessibility behavior, please opt in into the use of legacy accessibility features explicitly by setting this AppContext switch value to *true*. 
+
+__Added support for UI Automation patterns and properties__
+- [UIA](https://docs.microsoft.com/en-us/dotnet/framework/ui-automation/ui-automation-overview) [Expand/Collapse pattern](https://docs.microsoft.com/en-us/dotnet/framework/ui-automation/implementing-the-ui-automation-expandcollapse-control-pattern) to `T:System.Windows.Forms.ToolStripSplitButton`
+- UIA [ControlType](https://docs.microsoft.com/en-us/dotnet/framework/ui-automation/ui-automation-support-for-the-menubar-control-type) property value "MenuItem" to `T:System.Windows.Forms.ToolStripMenuItem`
+- UIA [Name](https://docs.microsoft.com/en-us/dotnet/api/system.windows.automation.automationelement.nameproperty?view=netframework-4.7) property and Expand/Collapse pattern to `T:System.Windows.Forms.ToolStripItem`
+- [AccessibleEvents](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.accessibleevents?view=netframework-4.7) indicating StateChange and NameChange when drop down is expanded or collapsed to `T:System.Windows.Forms.ToolStripDropDownItem`
+- UIA ControlType property value "Button" to `T:System.Windows.Forms.ToolStripDropDownButton`
+- UIA [TogglePattern](https://docs.microsoft.com/en-us/dotnet/api/system.windows.automation.togglepattern?view=netframework-4.7) to `T:System.Windows.Forms.DataGridViewCheckBoxCell`
+- UIA Expand/Collapse pattern to `T:System.Windows.Forms.ComboBox` 
+ 
+__Improvements to PropertyBrowser control__
+- Added UIA Expand/Collapse pattern, state and name change notifications and ControlType property with value "Button" to the "Details" button in the error dialog that is displayed when the user enters incorect value in the `T:System.Windows.Forms.PropertyGrid` control 
+- Changed [AccessibleRole](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.accessiblerole?view=netframework-4.7) of rows in PropertyGrid control from "Row" to "Cell", which maps to UIA ControlType "DataItem", in order to support appropriate keyboard shortcuts and Narrator announcements
+- Improved keyboard navigation between the grid and the ToolBar above it. Pressing "Shift-Tab" now selects the first ToolBar button, instead of the whole ToolBar
+
+__Ensure that OS-defined colors are used in High Contrast themes__
+
+_Note:_ 
+Windows10 had changed values for some high contrast system colors in the *Fall Creator's Update* release. Windows Forms Framework is based on Win32 framework, for he best experience, please run on the latest version of windows and opt in into the latest OS changes by adding app.manifest file in test application and uncommented following code:
+```
+    <!-- Windows 10 -->
+    <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}" />
+```
+
```

Comment:
This should probably go at the end of this section (after FlatStyle.System) rather than first.

NOTE: The Windows10 Creator's Update changed values for some high contrast system colors. Windows Forms Framework is based on Win32 framework. For the best experience, run on the latest version of Windows and opt in to the latest OS changes by adding an app.manifest file in a test application and uncommenting the following code: 

```xml 
<!-- Windows 10 --> 
<supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}" /> 
```

Reply:
This "note" applies to the whole section about high contrast themes, there will be more bugs described there.
Windows fixes I'm referring to are in RS3, in the upcoming release, not in the creator's update